src/celengine/meshmanager.cpp

```diff
@@ -282,7 +276,7 @@ ConvertTriangleMesh(const M3DTriangleMesh& mesh,
                 if (faceNormals[i].dot(faceNormals[k]) > 0.5f)
                     v += faceNormals[k];
             }
-            vertexNormals[i * 3 + 2] = v.normalized();
+            vertexNormals.push_back(v.normalized());
```

Comment:
probably emplace_back should be here to avoid an copy

Reply:
As the result of calling `normalized` is a temporary, this will call the rvalue overload of `push_back` and do a move. On the other hand, fixed size Eigen objects hold their data directly in an array member, so move and copy will end up doing the same thing.